### ISSUE TITLE
unused variable, false positive

### ISSUE THREAD

**Issue by larshp** [abaplint/abaplint#1634]
```abap
        DATA(ls_tadir) = cl_wb_object_type=>get_tadir_from_limu(
          p_object   = ls_list-object
          p_obj_name = ls_list-obj_name ).

        APPEND CORRESPONDING #( ls_tadir ) TO rt_list.
```

![image](https://user-images.githubusercontent.com/5888506/103340722-15b19f00-4a85-11eb-9dac-bf28259af4a3.png)

**Comment by larshp** [comment]
```abap
TYPES:
  BEGIN OF ty_result,
    ci_has_errors TYPE abap_bool,
  END OF ty_result.
DATA ls_tadir TYPE ty_result.
DATA rt_list TYPE STANDARD TABLE OF ty_result WITH EMPTY KEY.
APPEND CORRESPONDING #( ls_tadir ) TO rt_list.
```
